### PR TITLE
[WS-Passive-Scan] Add Private IP Disclosure

### PR DESCRIPTION
**Pull request by ManosMagnus**
This script scans websocket messages for private IP V4 addresses as well as Amazon EC2 private hostnames Private IP V4 such as: 10.x.x.x, 172.x.x.x, 192.168.x.x Amazon EC2 private hostname: such as ip-10-0-56-78

__________________________________________________________
TODOs:

- [x] Add help content
- [x] Based on  #2089

### CODE REVIEW

**Comment by ManosMagnus** [comment]
Help content was added, let me if you think that it is redundant to use examples and default values in help page.

File: addOns/websocket/src/main/javahelp/org/zaproxy/zap/extension/websocket/resources/help/contents/pscanrules.html [hunk 1]
```diff
@@ -11,5 +11,22 @@ <H1>WebSocket Passive Scan Rules</H1>
...
         Scripts which are included by default in the add-on and they implement the following WebSocket passive scan rules:
 
+        <h2 id="ip_add_disc">Private Address Disclosure</h2>
+        Checks incoming WebSocket message payload for inclusion of RFC 1918 IPv4 addresses as well as Amazon EC2 private hostnames (for example, ip-10-0-56-78). This information can give an attacker useful information about the IP address scheme of the internal network, and might be helpful for further attacks targeting internal systems. <br>
+
+        This passive scanner may generate false positives in the case of larger dotted numeric strings, such as vp09.02.51.10.01.09.16, where the latter 4 octets appear to be a RFC 1918 IPv4 address. After review an analyst can mark such alerts as False Positives in ZAP.<br>
+        <br>
+        <table border="1"  width = "200">
+            <tr><th>Use case</th><th>Output</th></tr>
+            <tr><td>10.255.255.255:</td><td>True Positive</td></tr>
```
Comment (kingthorin) [inline addOns/websocket/src/main/javahelp/org/zaproxy/zap/extension/websocket/resources/help/contents/pscanrules.html]: Should the trailing colon be included?

**Comment by ManosMagnus** [comment]
> Based on another branch/pr?

Right, reference was added in the description

File: addOns/websocket/src/main/javahelp/org/zaproxy/zap/extension/websocket/resources/help/contents/pscanrules.html (hunk 1 above)
Comment (ManosMagnus) [reply addOns/websocket/src/main/javahelp/org/zaproxy/zap/extension/websocket/resources/help/contents/pscanrules.html]: There wasn't necessary, so it is removed

**Comment by kingthorin** [comment]
Oops totally missed that, tks!

**Comment by ManosMagnus** [comment]
> Oops totally missed that, tks!

Actually was my fault :) I meant that it was added after your's mention

**Comment by ManosMagnus** [comment]
Rebased and ready for any further review

File: addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/private_id_disclosure.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        matches.forEach(function(evidence){
+            helper.newAlert()
+                .setRiskConfidence(RISK_LOW, CONFIDENCE_MEDIUM)
+                .setName("Private IP Disclosure via WebSocket (script)")
+                .setDescription("A private IP (such as 10.x.x.x, 172.x.x.x, 192.168.x.x)\
+ or an Amazon EC2 private hostname (for example, ip-10-0-56-78) has been found in the incoming\
+ WebSocket message. This information might be helpful for further attacks targeting\
+ internal systems.")
+                .setSolution("Remove the private IP address from the WebSocket messages.\
+  For comments, use JSP/ASP/PHP comment instead of HTML/JavaScript comment which can be\
```
Comment (thc202) [inline addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/private_id_disclosure.js]: Remove? Does not apply here.

**Comment by thc202** [comment]
> I think it looks good, not sure about calling the one column in the example table “output” but don’t have a better suggestion right this second either.

Outcome or Result?

The script name has a typo in IP. I think we should name them with proper case and use spaces, that will give a better script name in ZAP.

**Comment by ManosMagnus** [comment]
`Outcome` sounds good to me

File: addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/private_id_disclosure.js (hunk 2 above)
Comment (ManosMagnus) [reply addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/private_id_disclosure.js]: Fixed in 561fea7ce

**Comment by thc202** [comment]
Thanks!

**Comment by ManosMagnus** [comment]
Thank you!

**Pull request merged**